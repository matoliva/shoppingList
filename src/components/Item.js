import { useState } from "react";
import AmountInput from './AmountInput';

import './item.css';

const Item = (props) => {
    const [checked, setChecked] = useState(props.isSelected | false);

    const handleClick = () => {
        setChecked(!checked);
    }

    const handleQuantity = (updatedItem) => {
        const newState = { id: props.id, itemName: props.name, quantity: updatedItem };
        props.onSelectQuantity(newState);
    }

    return (
        <div className="item">
            <div className="item__desc">
                <input
                    className="item__desc__check" 
                    type="checkbox"
                    defaultChecked={checked}
                    onClick={ handleClick }></input>
                <span className={ checked ? 'item__check--checked' : '' } >{props.name}</span>
            </div>
            <AmountInput 
                quantity={props.quantity} 
                onSelectQuantity={handleQuantity} />
        </div>
    )
}

export default Item;