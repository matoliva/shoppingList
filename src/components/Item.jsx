import { useState } from "react";
import AmountInput from './AmountInput';
import PropTypes from 'prop-types';

import './item.css';

const Item = ({id, name, quantity = 0, isSelected = false, onSelectQuantity}) => {
    const [checked, setChecked] = useState(isSelected);

    const handleClick = () => {
        setChecked(!checked);
    }

    const handleQuantity = (updatedItem) => {
        const newState = { id: id, itemName: name, quantity: updatedItem };
        onSelectQuantity(newState);
    }

    return (
        <div className="item">
            <div className="item__desc">
                <input
                    className="item__desc__check" 
                    type="checkbox"
                    defaultChecked={checked}
                    onClick={ handleClick }></input>
                <span className={ checked ? 'item__check--checked' : '' } >{name}</span>
            </div>
            <AmountInput 
                quantity={quantity} 
                onSelectQuantity={handleQuantity} />
        </div>
    )
}

Item.propTypes = {
    id: PropTypes.number.isRequired, 
    name: PropTypes.string.isRequired, 
    quantity: PropTypes.number, 
    isSelected: PropTypes.bool,
    onSelectQuantity: PropTypes.func.isRequired
};

export default Item;