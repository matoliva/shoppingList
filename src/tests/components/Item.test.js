import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import Item from '../../components/Item';

describe('<Item />', () => {
    let props;
    let wrapper;

    beforeEach(() => {
        props = {
            key: 1,
            id: 1,
            onSelectQuantity: jest.fn(),
            name: 'test',
            quantity: 10,
            isSelected: false
        };  

        wrapper = shallow(<Item {...props} />);
    });

    test('should display <Item /> correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should span tag has "item__check--checked" class when the user click in checkbox input', () => {
        
        wrapper = shallow(<Item {...props} />);

        const inputCheckbox = wrapper.find('input');
        inputCheckbox.simulate('click');
        const spanElement = wrapper.find('span');

        expect(spanElement.hasClass('item__check--checked')).toBe(true);
    });

    test('should span tag has not "item__check--checked" class when the user click in a checked checkbox input', () => {
        const props = {
            key: 1,
            id: 1,
            onSelectQuantity: jest.fn(),
            name: 'test',
            quantity: 10,
            isSelected: true
        };

        wrapper = shallow(<Item {...props} />);

        const inputCheckboxElem = wrapper.find('input');

        inputCheckboxElem.simulate('click');
        
        const spanElement = wrapper.find('span');

        expect(spanElement.hasClass('item__check--checked')).toBe(false);
    });


});