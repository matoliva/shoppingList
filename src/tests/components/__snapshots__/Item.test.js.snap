// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> should display <Item /> correctly 1`] = `
<div
  className="item"
>
  <div
    className="item__desc"
  >
    <input
      className="item__desc__check"
      defaultChecked={false}
      onClick={[Function]}
      type="checkbox"
    />
    <span
      className=""
    >
      test
    </span>
  </div>
  <AmountInput
    onSelectQuantity={[Function]}
    quantity={10}
  />
</div>
`;
